packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -432,6 +461,134 @@ export class PartialSequenceLengths {
 		}
 	}
 
+	private static accumulateMoveClientOverlap(
+		partialLength: PartialSequenceLength,
+		overlapMoveClientIds: number[],
+		seglen: number,
+	) {
+		if (partialLength.overlapObliterateClients) {
+			for (const clientId of overlapMoveClientIds) {
+				const overlapClientNode = partialLength.overlapObliterateClients.get(clientId);
+				if (!overlapClientNode) {
+					partialLength.overlapObliterateClients.put(clientId, { clientId, seglen });
+				} else {
+					overlapClientNode.data.seglen += seglen;
+				}
+			}
+		} else {
+			partialLength.overlapObliterateClients = PartialSequenceLengths.getOverlapClients(
+				overlapMoveClientIds,
+				seglen,
+			);
+		}
+	}
+
+	static accumulateMoveOverlapForExisting(
+		segmentLen: number,
+		segment: ISegment,
+		firstGte: PartialSequenceLength,
+		moveClientOverlap: number[],
+	) {
+		const moveClientOverlapWithoutLocal = moveClientOverlap.filter(
+			(id) => id !== segment.clientId,
+		);
+
+		PartialSequenceLengths.accumulateMoveClientOverlap(
+			firstGte,
+			moveClientOverlapWithoutLocal,
+			segmentLen,
+		);
+
+		if (moveClientOverlap.length !== moveClientOverlapWithoutLocal.length) {
+			PartialSequenceLengths.accumulateMoveClientOverlap(
+				firstGte,
+				[segment.clientId],
+				segment.wasObliteratedOnInsert ? -segment.cachedLength : segmentLen,
+			);
+		}
+	}
+
+	// todo: maybe pass object for args here? would cause overhead
+	// using primitive types here makes it easy to mess up args
+	private static calculateAfterInsertion(
+		segment: ISegment,
+		segmentLen: number,
+		remoteObliteratedLen: number | undefined,
+		obliterateOverlapLen: number | undefined,
+		partials: PartialSequenceLengthsSet,
+		seq: number,
+		clientId: number,
+		removeClientOverlap: number[] | undefined,
+		moveClientOverlap: number[] | undefined,
+	) {
```

Comment:
`todo` is somewhat a question for reviewers -- would we care about the performance of constructing an object here? this function takes a lot of arguments